### DISCUSSION TITLE
Is it possible to globally override response data on all routes?

### DISCUSSION THREAD

**Discussion by SuperMasterBlasterLaser** [strapi/strapi#6393] [Legacy Topics]
Hello.

Lets say that I have currently Content-Type that returns data like this:

    [{"id":1,"name":"Restaurant 1","description":"This is restaurant 1","is_active":true,"created_at":"2020-05-27T16:00:47.614Z","updated_at":"2020-05-27T16:00:47.614Z"}]

Instead of json array, I want to make it respond with my custom wrapped body like this:

    {
         "status": "ok".
         "body": [
              /* Here goes my content-types  */
          ]
    } 

I know that I can override specific controller on model. However, I want to globally override response data on all possible routes.

Also i want to globally add additional data on response headers.

Is it possible to make global override for response of all routes in all controllers?

**Comment by lauriejim** [comment]
Hello! If you want to do that for all route you can create a custom middleware.
Here is the [documentation](https://strapi.io/documentation/v3.x/concepts/middlewares.html).
But be careful to create a condition for request that concern only generated API endpoint and not plugins endpoints.
Because this could break the admin if you change response for the admin.
By doing a recursive on `strapi.api` you will find all APIs and then `.config.routes` to find all routes.
And you will be able to update the ctx.body and ctx.headers of the response.
